fog/ledger/server/tests/store.rs

```diff
@@ -0,0 +1,331 @@
+        .expect("Error adding key image data to the enclave.");
+
+    // Set up the client's end of the encrypted connection.
+    let initiator = Start::new(responder_id.to_string());
+
+    let init_input = ClientInitiate::<X25519, Aes256Gcm, Sha512>::default();
+    let (initiator, auth_request_output) = initiator
+        .try_next(&mut rng, init_input)
+        .expect("Could not encrypt auth message.");
+
+    // Authenticate our "client" with the server.
+    let auth_message = attest::AuthMessage::from(auth_request_output);
+    let (client_auth_response, client_session) = enclave
+        .client_accept(auth_message.into())
+        .expect("Unable to connect a dummy \"client\" connection to the enclave.");
+
+    // We will need to double-convert, ClientAuthResponse -> AuthMessage ->
+    // AuthResponseOutput
+    let auth_message = attest::AuthMessage::from(client_auth_response);
+    // Initiator accepts responder's message.
+    let auth_response_event = AuthResponseInput::new(auth_message.into(), Verifier::default());
+    // Should be a valid noise connection at this point.
+    let (mut noise_connection, _verification_report) = initiator
+        .try_next(&mut rng, auth_response_event)
+        .expect("Could not get a noise connection and verification report from the initiator.");
+
+    //Construct our request.
+    let key_images_request = CheckKeyImagesRequest {
+        queries: vec![KeyImageQuery {
+            key_image: test_key_image.key_image,
+            start_block: 1,
+        }],
+    };
+    // Protobuf-encoded plaintext.
+    let message_encoded = mc_util_serial::encode(&key_images_request);
+    let ciphertext = noise_connection
+        .encrypt(&[], &message_encoded)
+        .expect("Failed to encrypt request from the client to the router.");
+    let msg: EnclaveMessage<ClientSession> = EnclaveMessage {
+        aad: vec![],
+        channel_id: client_session,
+        data: ciphertext,
+    };
+
+    // Decrypt and seal
+    let sealed_query = enclave
+        .decrypt_and_seal_query(msg)
+        .expect("Unable to decrypt and seal client message.");
+
+    let mut multi_query = enclave
+        .create_multi_key_image_store_query_data(sealed_query.clone())
+        .expect("Could not create multi key image store query data.");
+
+    let query = multi_query
+        .pop()
+        .expect("Query should have had one message");
+    println!("Nonce session on message is {:?}", query.channel_id);
+
+    // Get an untrusted query
+    let (
+        highest_processed_block_count,
+        last_known_block_cumulative_txo_count,
+        latest_block_version,
+    ) = {
+        let shared_state = shared_state.lock().expect("mutex poisoned");
+        (
+            shared_state.highest_processed_block_count,
+            shared_state.last_known_block_cumulative_txo_count,
+            shared_state.latest_block_version,
+        )
+    };
+
+    let untrusted_kiqr = UntrustedKeyImageQueryResponse {
```

Comment:
Please rename the `kiqr` to something like `untrusted_response`. The meaning `kiqr` isn't obvious.